cmd/jimmctl/cmd/role.go

```diff
@@ -0,0 +1,384 @@
+	modelcmd.ControllerCommandBase
+
+	store    jujuclient.ClientStore
+	dialOpts *jujuapi.DialOpts
+
+	name    string
+	newName string
+}
+
+// Info implements the cmd.Command interface.
+func (c *renameRoleCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:    "rename",
+		Purpose: "Rename a role.",
+		Doc:     renameRoleDoc,
+	})
+}
+
+// Init implements the cmd.Command interface.
+func (c *renameRoleCommand) Init(args []string) error {
+	if len(args) < 2 {
+		return errors.E("role name not specified")
+	}
+	c.name, c.newName, args = args[0], args[1], args[2:]
+	if len(args) > 0 {
+		return errors.E("too many args")
+	}
+	return nil
+}
+
+// Run implements Command.Run.
+func (c *renameRoleCommand) Run(ctxt *cmd.Context) error {
+	currentController, err := c.store.CurrentController()
+	if err != nil {
+		return errors.E(err, "could not determine controller")
+	}
+
+	apiCaller, err := c.NewAPIRootWithDialOpts(c.store, currentController, "", c.dialOpts)
+	if err != nil {
+		return err
+	}
+
+	params := apiparams.RenameRoleRequest{
+		Name:    c.name,
+		NewName: c.newName,
+	}
+
+	client := api.NewClient(apiCaller)
+	err = client.RenameRole(&params)
+	if err != nil {
+		return errors.E(err)
+	}
+
+	return nil
+}
+
+// newRemoveRoleCommand returns a command to Remove a role.
+func newRemoveRoleCommand() cmd.Command {
+	cmd := &removeRoleCommand{
+		store: jujuclient.NewFileClientStore(),
+	}
+
+	return modelcmd.WrapBase(cmd)
+}
+
+// removeRoleCommand Removes a role.
+type removeRoleCommand struct {
+	modelcmd.ControllerCommandBase
+	out cmd.Output
+
+	store    jujuclient.ClientStore
+	dialOpts *jujuapi.DialOpts
+
+	name  string
+	force bool
+}
+
+// Info implements the cmd.Command interface.
+func (c *removeRoleCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:    "remove",
+		Purpose: "Remove a role.",
+		Doc:     removeRoleDoc,
+	})
+}
+
+// Init implements the cmd.Command interface.
+func (c *removeRoleCommand) Init(args []string) error {
+	if len(args) < 1 {
+		return errors.E("role name not specified")
+	}
+	c.name, args = args[0], args[1:]
+	if len(args) > 0 {
+		return errors.E("too many args")
+	}
+	return nil
+}
+
+// SetFlags implements Command.SetFlags.
+func (c *removeRoleCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.CommandBase.SetFlags(f)
+	c.out.AddFlags(f, "smart", map[string]cmd.Formatter{
+		"smart": cmd.FormatSmart,
+	})
+	f.BoolVar(&c.force, "y", false, "delete role without prompt")
+}
+
+// Run implements Command.Run.
+func (c *removeRoleCommand) Run(ctxt *cmd.Context) error {
+	currentController, err := c.store.CurrentController()
+	if err != nil {
+		return errors.E(err, "could not determine controller")
+	}
+
+	if !c.force {
+		reader := bufio.NewReader(ctxt.Stdin)
+		// Using Fprintf over c.out.write to avoid printing a new line.
+		_, err := fmt.Fprintf(ctxt.Stdout, "This will also delele all associated relations.\nConfirm you would like to delete role %q (y/N): ", c.name)
```

Comment:
```suggestion
		_, err := fmt.Fprintf(ctxt.Stdout, "This will also delete all associated relations.\nConfirm you would like to delete role %q (y/N): ", c.name)
```